src/theory/arith/theory_arith_private.cpp

```diff
@@ -3938,15 +3938,36 @@ Node TheoryArithPrivate::branchIntegerVariable(ArithVar x) const {
   TNode var = d_partialModel.asNode(x);
   Integer floor_d = d.floor();
 
-  //Node eq = Rewriter::rewrite(NodeManager::currentNM()->mkNode(kind::EQUAL, var, mkRationalNode(floor_d+1)));
-  //Node diseq = eq.notNode();
+  if (options::cubeTest())
+  {
+    Trace("integers") << "cube test enabled"  << endl;
+    Integer ceil_d = d.ceiling();
+    Rational f = r - floor_d;
+    // Multiply by -1 to get abs value.
+    Rational c = (r - ceil_d) * (-1); 
+ 
+    Integer nearest;
+    (c > f) ? nearest = floor_d : nearest = ceil_d;
 
-  Node ub = Rewriter::rewrite(NodeManager::currentNM()->mkNode(kind::LEQ, var, mkRationalNode(floor_d)));
-  Node lb = ub.notNode();
+    Node ub = Rewriter::rewrite(NodeManager::currentNM()->mkNode(kind::LEQ, var, mkRationalNode(nearest - 1)));
+    Node lb = Rewriter::rewrite(NodeManager::currentNM()->mkNode(kind::GEQ, var, mkRationalNode(nearest + 1)));
+    Node lem = NodeManager::currentNM()->mkNode(kind::OR, ub, lb);
+    Node eq = Rewriter::rewrite(NodeManager::currentNM()->mkNode(kind::EQUAL, var, mkRationalNode(nearest)));
+    Node literal = d_containing.getValuation().ensureLiteral(eq);
```

Comment:
Can you give a few comments here (one or two lines would suffice) to state the intuition behind this code.

Reply:
> I think these are equivalent. x >= c+1 is equivalent to ~(x <= c ) right (for integers)? I prefer leaving the code unchanged here since its clearer that this is indeed a proper split.

These comments and this code is unchanged from the original.